backend/internal/service/store/user.go

```diff
@@ -89,3 +89,81 @@ func (mh *mongoHandler) SetWalletBalance(userName string, newBalance int) error
 	}
 	return nil
 }
+
+// Not Tested.
+func (mh *mongoHandler) AddWalletTransaction(userName string, walletTxID string, stockID string, is_debit bool, amount int, timeStamp int64) error {
+	var walletTx models.WalletTransaction = models.WalletTransaction{
+		UserName:   userName,
+		WalletTxID: walletTxID,
+		StockID:    stockID,
+		Is_debit:   is_debit,
+		Amount:     amount,
+		TimeStamp:  timeStamp,
+	}
+
+	// Add to 'wallet_transactions' collection
+	collection := mh.client.Database("day-trading-app").Collection("wallet_transactions")
+	_, err := collection.InsertOne(context.Background(), bson.M{"stock_tx_id": walletTx})
+	if err != nil {
+		return err
+	}
+
+	// * Add to user's entry in 'users' collection *
+
+	// Access the collection where user data is stored
+	collection = mh.client.Database("day-trading-app").Collection("users")
+
+	// Find the user by their username
+	var user models.User
+	err = collection.FindOne(context.Background(), bson.M{"user_name": userName}).Decode(&user)
+	if err != nil {
+		return err
+	}
+
+	user.WalletTxns = append(user.WalletTxns, walletTx)
```

Comment:
you can use 
```
_, err = collection.UpdateOne(context.Background(),
        bson.M{"user_name": user.Username}, 
        bson.M{"$addToSet": bson.M{"WalletTxns": WalletTx}},
    )
```
insert into the database directly, instead of append to the list then replace the old with new.